deeplake/auto/unstructured/yolo/yolo.py

```diff
@@ -0,0 +1,329 @@
+
+        self.allow_no_annotation = allow_no_annotation
+        self.verify_class_names = verify_class_names
+        self.creds = creds
+        self.image_creds_key = image_creds_key
+        self.inspect_limit = inspect_limit
+
+        self.data = YoloData(
+            self.data_directory,
+            creds,
+            self.annotations_directory,
+            self.class_names_file,
+        )
+        self._validate_data()
+
+        # Create a separate list of tuples with the intestion data (img_fn, annotation_fn).
+        # We do this in advance so missing files are discovered before the ingestion process.
+        self._create_ingestion_list()
+
+        self._validate_ingestion_data()
+
+        self._initialize_params(
+            image_params or {}, label_params or {}, coordinates_params or {}
+        )
+        self._validate_image_params()
+
+    def _parse_coordinates_type(self):
+        """Function inspects up to inspect_limit annotation files in order to infer whether they are polygons or bounding boxes"""
+
+        # If the htype or name of the coordinates is not specified (htype could be bbox or polygon), auto-infer it by reading some of the annotation files
+        if (
+            "htype" not in self.coordinates_params.keys()
+            or "name" not in self.coordinates_params.keys()
+        ):
+
+            # Read the annotation files assuming they are polygons and check if there are any non-empty annotations without 4 coordinates
+            coordinates_htype = "bbox"  # Initialize to bbox and change if contradicted
+            coordinates_name = "boxes"  # Initialize to boxes and change if contradicted
+            count = 0
+            while count < min(self.inspect_limit, len(self.ingestion_data)):
+                fn = self.ingestion_data[count][1]
+                if fn is not None:
+                    _, coordinates = self.data.read_yolo_coordinates(fn, is_box=False)
+                    for c in coordinates:
+                        coord_size = c.size
+                        if coord_size > 0 and coord_size != 4:
+                            coordinates_htype = "polygon"
+                            coordinates_name = "polygons"
+
+                            count = (
+                                self.inspect_limit + 1
+                            )  # Set this to exit the while loop
+                            break
+
+                        ## TODO: Add fancier math to see whether even coordinates with 4 elements could be polygons
+                count += 1
+
+        if "htype" not in self.coordinates_params.keys():
+            self.coordinates_params["htype"] = coordinates_htype
+
+        if "name" not in self.coordinates_params.keys():
+            self.coordinates_params["name"] = coordinates_name
+
+    def _initialize_params(self, image_params, label_params, coordinates_params):
+        image_params_updated = DEFAULT_IMAGE_TENSOR_PARAMS.copy()
```

Comment:
These parts may be simplified like `self.image_params = {**DEFAULT_IMAGE_TENSOR_PARAMS, **image_params}`. Effect if the same, only items present in the second dictionary will be overwritten over the default ones.

Reply:
fixed